lib/fs.js

```diff
@@ -283,35 +190,41 @@ fs.access = function(path, mode, callback) {
     throw new errors.TypeError('ERR_INVALID_CALLBACK');
   }
 
-  if (handleError((path = getPathFromURL(path)), callback))
-    return;
+  handleError(path = getPathFromURL(path));
```

Comment:
Hm, my first thought when I saw the change to handle errors sync was that it is a good idea but after having a second thought I am not so sure about that anymore. The path could definitely be passed in by users and therefore handling the error in the callback would probably be the best thing to stick to?

Reply:
Currently, argument validation error handling is inconsistent, with some errors resulting in a throw and others being passed to the callback. I'm good with passing them to the callback, it just needs to be consistent (e.g. all validation errors always going to the callback or always throwing).

Reply:
I agree that they should be consistent but as long as it is not a "programmers" error I think it should land in the callback.